devlab/tests/base.py

```diff
@@ -252,6 +252,9 @@ def user_has_not_primary_tenants(self, user_name):
 
     def check_vm_state(self, srv_id):
         srv = self.novaclient.servers.get(srv_id)
+        if srv.status == 'ERROR':
+            msg = 'VM with id {0} was spawned in error state'
+            raise RuntimeError(msg.format(srv_id))
```

Comment:
I'd suggest using custom exception type and handle it appropriately in the calling code.